### ISSUE TITLE
Crash when using TwilioVoice.isAudioEnabled

### ISSUE THREAD

**Issue by mahyarr** [twilio/voice-quickstart-swift#94]
### Steps to Reproduce

Setting ```TwilioVoice.isAudioEnabled``` to true or false sometimes crashes the app

### Steps to Reproduce

1. Make a call

#### Code

```swift
TwilioVoice.isAudioEnabled = true
```

#### Reproduces How Often

Almost 50% of the time

#### Logs

```
PJSIP(1): pjsua_aud.c  .Unable to open sound device: Unknown error 506637 [status=506637]
Assertion failed: (status == PJ_SUCCESS), function +[TwilioVoice enableAudio]_block_invoke, file /Users/ibadmin/jenkins/workspace/voice-ios-snapshot/TwilioVoiceClient/TwilioVoiceClient/Sources/TwilioVoice.mm, line 299.
```

#### Voice iOS SDK

2.0.0 GA

#### Xcode

9.2

#### iOS Version

11.2.1

#### iOS Device

iPhone 7 Plus

**Comment by idelgado** [comment]
@mahyarr is this specifically when using it with CallKit? We have a number of tests that toggle this during the connecting, connected, and disconnecting phase so i'm trying to get better coverage in this area to reproduce this locally.

**Comment by mahyarr** [comment]
Yes, I'm using it with CallKit and I've put the calls where you showed in the sample project.

The odd thing is the app has stopped crashing since last night without me changing anything in the code.

**Comment by idelgado** [comment]
We are adding a retry mechanism to re-attempt enabling the audio device when it fails to open. Unfortunately we have not been able to reproduce this locally but we've seen similar retry code in the WebRTC code base when enabling the audio device. I'll post an update once there is an artifact available with this change.

**Comment by mahyarr** [comment]
Thanks @idelgado appreciate your quick reaction to this issue

**Comment by idelgado** [comment]
We released a new version of the SDK, TwilioVoice 2.0.1. This disables assertions in release builds (we identified this as an issue in our build scripts last week) and also retries opening the audio device which should help resolve the issue you reported here.

**Comment by bchen-twilio** [comment]
Closing this issue for now. Please feel free to open new issues.
Thanks.
